### PR TITLE
Added Cyclic Permutation

### LINKED ISSUE #1298: Find Cyclic Permutations of a number in C, C++, Java, Python, Go

### PR DESCRIPTION
**Pull request by MastersAbh**
Fixes #1298

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- Added Implementation for cyclic permutation of a number in 5 languages

#### Languages Used:
- C
- C++
- Java
- Python
- Go

#### Files Added:
- Cyclic_Permutation.c
- Cyclic_Permutation.cpp
- Cyclic_Permutation.py
- Cyclic_Permutation.java
- Cyclic_Permutation.go

### CODE REVIEW

File: Cyclic_Permutation/Cyclic_Permutation.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+'''
+        CYCLIC PERMUTATIONS
```
Comment (jainaman224) [inline Cyclic_Permutation/Cyclic_Permutation.py]: Fix Indentation.

**Comment by MastersAbh** [comment]
Done @jainaman224 . Please review.

**Comment by MastersAbh** [comment]
@rishabhgarg25699 please have a look at this PR.

File: Cyclic_Permutation/Cyclic_Permutation.cpp [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        if (res == n) {
+            break;
+        }
+    }
+}
+
+int main() {
+    cout<<"Enter N:";
+    int n;
+    cin>>n;
```
Comment (jainaman224) [inline Cyclic_Permutation/Cyclic_Permutation.cpp]: Space around `<<`;

File: Cyclic_Permutation/Cyclic_Permutation.cpp [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    return ans;
+}
+
+void cyclic(int n) {
+    int length = getDigits(n);
+    int res = n;
+    while (1) {
+        cout<<res<<endl;
+        int r = res % 10;
+        int d = res / 10;
```
Comment (jainaman224) [inline Cyclic_Permutation/Cyclic_Permutation.cpp]: Use descriptive name for `r`, `d`.

File: Cyclic_Permutation/Cyclic_Permutation.cpp (hunk 2 above)
Comment (MastersAbh) [reply Cyclic_Permutation/Cyclic_Permutation.cpp]: Hi. @marwahmanbir had asked to remove spaces around << and >> for cin and cout statements in PRs.

Comment (MastersAbh) [reply Cyclic_Permutation/Cyclic_Permutation.cpp]: What should I do @jainaman224 ?

**Comment by MastersAbh** [comment]
Done, Please review @jainaman224

**Pull request merged**